### PR TITLE
Port the pocket beam underbarrel attachment from RUTGMC

### PR DESCRIPTION
**Pull request by 00ze-cyclone**
## About The Pull Request

Add the RUTGMC pocket beam underbarrel attachment with some modifications.
It use TE cells like all other lasers, has 3 firemodes, all of them have 6 shots and a 2 seconds fire delay:
Piercing: fire a laser that goes through mob, 50 damage, 10 AP, 1 sunder
Ricochet: literally just lasniper ricochet (but with less ammo and lower rof)
Impact: 3 beams, no damage or sunder, each beam push xenos away

You can switch between the modes when the pocket beam is deattached so it's not 3 attachments in one at all times, it do take less time to attach and deattach it however so you have some incentive to try to use more than one mode during the round.
Basically any gun that can equip a masterkey can equip the pocket beam, laser weapons can't use and will NEVER be able to use it because it would be additional firemodes for too low of a cost.
LASER LASER LASER LASER LASER LASER LASER LASER

## Why It's Good For The Game

More options for underbarrels, tried to make the firemodes not overlap with existing underbarrels.
Also YOU WILL USE LASERS EVEN IF YOU ARE USING A BALLISTIC GUN NOW!

## Changelog

:cl:homexp13, 00ze-cyclone
add: Pocket beam underbarrel
/:cl:

### CODE REVIEW

**Comment by Lauwma** [comment]
This is straightup an unconditionnal dps upgrade unlike most underbarrel lol

**Comment by yyzsong** [comment]
>3 seconds of shatter underbarrel

chat is this real

**Comment by 00ze-cyclone** [comment]
> This is straightup an unconditionnal dps upgrade unlike most underbarrel lol. It gives easy access to shatter or straight up +25 dps. I already see the ar18 rampage that would cause



> > 3 seconds of shatter underbarrel
> 
> chat is this real

Okay yeah after a little thinking shatter on an underbarrel is too much, going to get something to replace it and reopen that PR.

**Comment by 00ze-cyclone** [comment]
> [PMC-Unga-Marines/Unga-Marines#378](https://github.com/PMC-Unga-Marines/Unga-Marines/pull/378) also port this as sprite fix

we'll probably never need that sprite since laser linking got shot dead and will likely never come back but just in case I added it.
Thank for the help

**Comment by Kuro020** [comment]
This is way too strong to be a underbarrel
